### PR TITLE
Windows

### PR DESCRIPTION
**Pull request by MisterDA**
There are various places where I think that quoting the parameter with `%S` instead of `%s` is unnecessary.

### CODE REVIEW

**Comment by talex5** [comment]
I'm not keen on removing the quotes. For example `Invalid job ID ""` is much clearer than `Invalid job ID`.

**Comment by MisterDA** [comment]
I've added the quotes back, I think we can safely merge the three patches.

File: lib/job.ml [hunk 1]
```diff
@@ -61,7 +61,7 @@ let jobs_dir = lazy (Disk_store.state_dir "job")
 let log_path job_id =
   let open Astring in
   let jobs_dir = Lazy.force jobs_dir in
-  match String.cuts ~sep:"/" job_id with
+  match String.cuts ~sep:Fpath.dir_sep job_id with
```
Comment (talex5) [inline lib/job.ml:64]: This doesn't seem right. The syntax of job IDs shouldn't depend on the platform. Probably `id_of_path` should be changed so they always use `/`.

File: lib/job.ml [hunk 2]
```diff
@@ -71,7 +71,7 @@ let log_path job_id =
       | Ok false -> Error (`Msg (Fmt.str "Job log %a does not exist" Fpath.pp path))
       | Error _ as e -> e
     end
-  | _ -> Error (`Msg (Fmt.str "Invalid job ID %S" job_id))
+  | _ -> Error (`Msg (Fmt.str "Invalid job ID \"%s\"" job_id))
```
Comment (talex5) [inline lib/job.ml:74]: Is this change intentional?

File: lib/job.ml (hunk 1 above)
Comment (MisterDA) [reply lib/job.ml]: Changed to use `Fpath.segs` then re-concatenating the results with `/`.

File: lib/job.ml (hunk 2 above)
Comment (MisterDA) [reply lib/job.ml]: I took to literally your "I'm not keen on removing the quotes". Removed the commit.

**Comment by talex5** [comment]
Thanks!
